linux/LibSensors.c

```diff
@@ -261,6 +252,38 @@ void LibSensors_getCPUTemperatures(CPUData* cpus, unsigned int existingCPUs, uns
       goto out;
    }
 
+   /* Check AMD Zen CPUs packages, and mirror temps across packages */
+   if (coreTempCount > 0 && coreTempCount != existingCPUs) {
+      double temp[coreTempCount];
+      unsigned int count = 0;
+      unsigned int coresInDie = existingCPUs / coreTempCount;
+
+      // Find package temps
+      for (unsigned int i = 1; i <= existingCPUs; i++) {
+         if (!isnan(data[i])) {
+            temp[count] = data[i];
+            count++;
+            if (count == coreTempCount) {
+               break;
+            }
+         }
+      }
+
+      /* Returns to conditional checking if no temp is found  */
+      if (!isnan(temp[0])) {
+         // Set Temperature
+         data[0] = temp[0];
+         for (unsigned int die = 0; die < coreTempCount; die++) {
+            for (unsigned int i = 1; i <= coresInDie; i++) {
+               data[(die * coresInDie) + i] = temp[die];
+            }
+         }
+
+         /* No further adjustments */
+         goto out;
```

Comment:
This should be one level up, because the special-casing should end here regardless of a global temperature found or not.

Reply:
I think that the goto statement should stay here and a fallback conditional should be created for the case where coreTempCount is 1 and data[0] has the only value in it. 

This check can be added into the current conditional statement, however, from looking at all the other conditional statements, we should have that moved out into its own check so that it remains a fallback and is not coupled with the section specified for AMD Zen CPU temperature reporting. (change idea added below)

I may be needlessly worried about a impossible or niche hardware/platform config. 

```suggestion

         /* No further adjustments */
         goto out;
      }
      /* AMD Zen CPU can report coreTempCount as 1 - use fallback conditional */
   }

   /* Fallback for single temperature count reporting */
   if (coreTempCount == 1 && !isnan(data[0])) {
      for (unsigned int i = 1; i <= existingCPUs; i++) {
         data[i] = data[0];

      /* No further adjustments */
      goto out;
   }

```